### PR TITLE
🐛 Fix bug with partnername in summary list

### PR DESCRIPTION
**Pull request by Sven65**
## Explain the changes you’ve made

In accordance with #CU-1ubxk7k, I've fixed a bug where `#partnerName` would not be replaced in summary lists.

## Explain your solution

I've added the additional logic needed to replace the question category descriptions. I've also updated the tests and made a small change to the typings to better match the data we've got.

## How to test

Concrete example:

1. Checkout this branch
2. Get a case with co-applicant
3. Submit up until the first summary list (probably incomes)
4. See that it says `Inkomst [name]` and `Inkomst [partner name]` instead of `#firstName` and `#partnerName`

## Tested environments

- [] Storybook on a iOS device/simulator.
- [x] Building the Application on a iOS device/simulator.
- [] Building the Application on a Android device/simulator.

---

## Checklist for reviewers

Here are a few points that are useful for reviewers to check.

The short list of the (maybe) most important points are:

- Does this PR work as expected?
- Does the PR properly only concern a single feature/bug/thing?
- Is the code clear and concise?
- Will the changes impact existing functionality?
- Are there any linting errors (ESLint) that should be resolved?
- Do the automatic GitHub checks pass?

However it is good if a reviewer can be as thorough as possible,
within reasonable limits of course.

### Functionality

- Does this PR work as expected?
- If this is a new feature, does it add value, or is it a sign of feature-creep?
- Will the changes impact existing functionality?
- Does the changes create inconsistencies somewhere else?
- Does the code work on different platforms if applicable?
- How does the code handle if unexpected values are passed to functions etc.?

### Readability, syntax, formatting, style

- Is the code clear and concise?
- Has styling rules (Prettier) been applied to included files?
- Are functions/variables/etc. aptly named?

### Design

- Is the code properly planned and designed?
- Is code organized/separated in a logical fashion?
- Does the code follow clean code practices to a reasonable degree?
- Is the code self-documenting enough?
- Is the code in sync with existing patterns/technologies?
- Does the code reuse existing functionality where applicable?
- Is the code designed with reusability in mind?
- Does the code have hard-coded values that can be refactored?

### Patterns, idioms, tests, best practices

- Are there any linting errors (ESLint) that should be resolved?
- Is the code properly typed (TypeScript)?
- Does the code utilize language features to their full potential?

### Debugging, testing, configuring

- Is appropriate logging used?
- Have tests been added/modified to accommodate the changes?
- Is the feature sufficiently configurable in case tweaks are required in the future?

### Performance, reliability, scalability

- Is the code well optimized?
- Does the code scale?
- Does the code handle failure scenarios?

### Security

- Is there any obvious potential for exploits and security threats?
- Does code check proper authorization where applicable?

### Etiquettes

- Are commits atomic (does only one thing per commit)?
- Does the PR properly only concern a single feature/bug/thing?
- Are commit messages well-written?
- Do the automatic GitHub checks pass?

### CODE REVIEW

**Comment by helsingborgsstad** [comment]
Task linked: [CU-1ubxk7k Bugg: Logik för att ersätta #PARTNERNAME till verkliga namnet på partnern. (Summary list)](https://app.clickup.com/t/1ubxk7k)

**Comment by sebastianthulin** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 #### 1 Issues 
* Code issue 1 Code issues 

 [Details](https://app.embold.io/organization/gh/helsingborg-stad/repositories/fedfce3fd8695f238dea9cb00bd37ce7/pull-request/617)

**Review by dannil76 — APPROVED** [review]
Looks right to me

File: source/containers/Form/hooks/test/textReplacement.test.ts [hunk 1]
```diff
@@ -97,6 +107,9 @@ const doTest = (
   expect(res[0].description).toBe(expected);
   expect(res[0].questions?.[0].label).toBe(expected);
 
+  expect(res[0]?.questions?.[1].categories?.[0].description).toBe(expected);
```
[the remark this answers is not in the document]
Comment (dannil76) [reply source/containers/Form/hooks/test/textReplacement.test.ts:110]: Maria would like to see this resolved now. Can we live with this @karlknowit

[the remark this answers is not in the document]
Comment (Sven65) [reply source/containers/Form/hooks/test/textReplacement.test.ts:110]: I'm unsure if we've got any other way of solving it, if we do, we'll sort it out later.

**Pull request merged**